### PR TITLE
Issue 5033: (SLTS) - Read ChunkedSegmentStorageConfig from properties.

### LINKED ISSUE #5033: SLTS - Ability to initialize ChunkedSegmentStorageConfig values from config file or properties.
**Problem description**
We need ability to initialize ChunkedSegmentStorageConfig values from config file or properties.

**Problem location**
ChunkedSegmentStorageConfig 

**Suggestions for an improvement**
Read values from config file or properties and use them during initialization.

### PR DESCRIPTION
**Pull request by sachin-j-joshi**
Ability to initialize ChunkedSegmentStorageConfig values from config file or properties.
Signed-off-by: Sachin Joshi <sachin.joshi@emc.com>

**Change log description**  
Ability to initialize ChunkedSegmentStorageConfig values from config file or properties.

**Purpose of the change**  
Fixes #5033  SLTS - Ability to initialize ChunkedSegmentStorageConfig values from config file or properties.

**What the code does**  
Read values from config file or properties and use them during initialization.

**How to verify it**  
Clean build. Manual testing.

### CODE REVIEW

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorageConfig.java [hunk 1]
```diff
@@ -35,6 +48,8 @@
             .appendsDisabled(false)
             .build();
 
+    static final String COMPONENT_CODE = "pravegaservice";
```
Comment (eolivelli) [inline segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorageConfig.java:51]: Is this right?
What about 'storage' ?

File: bindings/src/main/java/io/pravega/storage/filesystem/FileSystemSimpleStorageFactory.java [hunk 2]
```diff
@@ -13,16 +13,19 @@
...
 import io.pravega.segmentstore.storage.chunklayer.ChunkedSegmentStorageConfig;
+import lombok.AllArgsConstructor;
 import lombok.NonNull;
-import lombok.RequiredArgsConstructor;
 
 import java.util.concurrent.ExecutorService;
 
 /**
- * Factory for FileSystem {@link Storage} implemented using {@link ChunkedSegmentStorage} and {@link FileSystemChunkStorage}.
+ * Factory for ExtendedS3 {@link Storage} implemented using {@link ChunkedSegmentStorage} and {@link FileSystemChunkStorage}.
```
Comment (andreipaduroiu) [inline bindings/src/main/java/io/pravega/storage/filesystem/FileSystemSimpleStorageFactory.java:22]: revert this

File: bindings/src/main/java/io/pravega/storage/filesystem/FileSystemSimpleStorageFactory.java [hunk 3]
```diff
@@ -13,16 +13,19 @@
...
  */
-@RequiredArgsConstructor
+@AllArgsConstructor
```
Comment (andreipaduroiu) [inline bindings/src/main/java/io/pravega/storage/filesystem/FileSystemSimpleStorageFactory.java:24]: And this.

File: bindings/src/main/java/io/pravega/storage/hdfs/HDFSSimpleStorageFactory.java [hunk 4]
```diff
@@ -13,18 +13,23 @@
...
 import lombok.NonNull;
-import lombok.RequiredArgsConstructor;
 
 import java.util.concurrent.Executor;
 
 /**
  * Factory for HDFS {@link Storage} implemented using {@link ChunkedSegmentStorage} and {@link HDFSChunkStorage}.
  */
-@RequiredArgsConstructor
+@AllArgsConstructor
```
Comment (andreipaduroiu) [inline bindings/src/main/java/io/pravega/storage/hdfs/HDFSSimpleStorageFactory.java:24]: I don't see why this is needed. RequiredArgsConstructor will generate the constructor for all final members.

File: config/config.properties [hunk 5]
```diff
@@ -106,6 +106,47 @@ pravegaservice.dataLog.impl.name=BOOKKEEPER
 # pravegaservice.storage.layout=ROLLING_STORAGE
 # NOTE: ChunkedSegmentStorage is an experimental feature. Its behavior and/or API should be expected to be in flux until fully released.
 
+# Size of chunk in bytes above which it is no longer considered a small object.
+# For small source objects, concat is not used.
+# Valid values: non-negative long
+# Default value: 0
+# storage.concat.size.bytes.min=0
+
+# Size of chunk in bytes above which it is no longer considered for concat.
```
Comment (andreipaduroiu) [inline config/config.properties:115]: Please rephrase this to be refer to the `max` argument.

File: config/config.properties [hunk 6]
```diff
@@ -106,6 +106,47 @@ pravegaservice.dataLog.impl.name=BOOKKEEPER
...
+# Default value: 1024 (1 K)
+# storage.readindex.chunksPerSegment.max=1024
+
+# Max number of indexed chunks to keep in cache.
+# Valid values: non-negative integer
+# Default value: 16384 (16 K)
+# storage.readindex.chunks.max=16384
+
+# The maximum size of a single Segment Chunk in Storage for metadata segments.
+# Valid values: non-negative long
```
Comment (andreipaduroiu) [inline config/config.properties:141]: Please be explicit about the bounds. A long can be a signed long or unsigned long. In this case, if I remember, the upper bound is 2^62, isn't it?

Please fix this for the other config descriptions too.

File: config/config.properties [hunk 7]
```diff
@@ -106,6 +106,47 @@ pravegaservice.dataLog.impl.name=BOOKKEEPER
...
+# Default value: 4611686018427387903
+# storage.metadata.rollover.size.bytes.max=4611686018427387903
+
+# Whether the append functionality is disabled.
+# Valid values: true, false
+# Default value: false
+# storage.appends.disable=false
```
Comment (andreipaduroiu) [inline config/config.properties:148]: For someone who is not really familiar with this, it makes more sense to name this `enabled` rather than `disabled`.

Comment (sachin-j-joshi) [reply config/config.properties]: done

File: config/config.properties (hunk 6 above)
Comment (sachin-j-joshi) [reply config/config.properties]: done

File: config/config.properties (hunk 5 above)
Comment (sachin-j-joshi) [reply config/config.properties]: fixed

File: bindings/src/main/java/io/pravega/storage/hdfs/HDFSSimpleStorageFactory.java (hunk 4 above)
Comment (sachin-j-joshi) [reply bindings/src/main/java/io/pravega/storage/hdfs/HDFSSimpleStorageFactory.java]: fixed

File: bindings/src/main/java/io/pravega/storage/filesystem/FileSystemSimpleStorageFactory.java (hunk 2 above)
Comment (sachin-j-joshi) [reply bindings/src/main/java/io/pravega/storage/filesystem/FileSystemSimpleStorageFactory.java]: fixed.

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorageConfig.java (hunk 1 above)
Comment (sachin-j-joshi) [reply segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorageConfig.java]: fixed. Thanks
